### PR TITLE
Marker with yaxis

### LINKED ISSUE #2738: Marker: It is not possible to define a marker on the y2 axis
It looks to be easy to fix.

### PR DESCRIPTION
**Pull request by vallsv**
This PR allows to define a marker with an yAxis (left or right).

It introduce an `yaxis` argument to the public API `addMarker`, `addXMarker`, `addYMarker` of the `PlotWidget`. Plus internal changes.

Closes #2738

### CODE REVIEW

File: silx/gui/plot/items/marker.py [hunk 1]
```diff
@@ -138,6 +140,23 @@ def setPosition(self, x, y):
             self._x, self._y = x, y
             self._updated(ItemChangedType.POSITION)
 
+    def getYAxis(self):
```
Comment (t20100) [inline silx/gui/plot/items/marker.py]: There is a `YAxisMixIn`which provides that.

File: silx/gui/plot/items/marker.py [hunk 2]
```diff
@@ -54,6 +54,7 @@ def __init__(self):
         self._text = ''
         self._x = None
         self._y = None
+        self._yAxis = None
```
Comment (t20100) [inline silx/gui/plot/items/marker.py]: That's probably what makes CI failing.
Better to use `YAxisMixIn`.

**Comment by t20100** [comment]
Made a few comments, from a look at the code, I don't see what can get wrong in `_updateMarkers`

File: silx/gui/plot/PlotWidget.py [hunk 3]
```diff
@@ -1436,18 +1445,28 @@ def addMarker(self, x, y, legend=None,
...
+                   symbol, constraint,
+                   yaxis=None):
         """Common method for adding point, vline and hline marker.
 
         See :meth:`addMarker` for argument documentation.
         """
         assert (x, y) != (None, None)
 
+        if yaxis is None:
+            # FIXME: The default None argument have be removed at one point
```
[the remark this answers is not in the document]
Comment (vallsv) [reply silx/gui/plot/PlotWidget.py]: You mean, in this PR?

Cause I am afraid that this private `_addMarker` is used by other applications.

**Comment by vallsv** [comment]
Looks good

File: silx/gui/plot/PlotWidget.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (t20100) [reply silx/gui/plot/PlotWidget.py]: I should not, this is a common implementation for the `add*Marker` methods, all its features are available through those methods... and it's private.

**Pull request merged**